Fabric.Identity.API/Configuration/IdentityProviderSearchSettings.cs

```diff
@@ -1,9 +1,42 @@
-﻿namespace Fabric.Identity.API.Configuration
+﻿using System.Threading.Tasks;
+using Fabric.Identity.API.Extensions;
+using Fabric.Identity.API.Services;
+
+namespace Fabric.Identity.API.Configuration
 {
     public class IdentityProviderSearchSettings
     {
-        public string BaseUrl { get; set;  }
+        /// <summary>
+        /// NOTE: Do not use this for retrieving the IdPSS URL. Instead, use <code>GetEffectiveBaseUrl</code>.
+        /// </summary>
+        public string BaseUrl { get; set; }
         public string GetUserEndpoint { get; set; }
         public bool IsEnabled { get; set; }
+
+        private static string _effectiveBaseUrl;
+
+        public async Task<string> GetEffectiveBaseUrl(IAppConfiguration appConfig)
+        {
+            if (string.IsNullOrEmpty(_effectiveBaseUrl))
+            {
+                _effectiveBaseUrl = await GetBaseUrl(appConfig);
+            }
+            return _effectiveBaseUrl;
+        }
+
+        private static async Task<string> GetBaseUrl(IAppConfiguration appConfig)
```

Comment:
This can also be an instance method.

Reply:
Resharper gave me a suggestion that it could be made static.